### PR TITLE
Feature/ipehr 168 adl14 api get list

### PR DESCRIPTION
**Pull request by Setest**

### CODE REVIEW

File: src/pkg/api/api.go [hunk 1]
```diff
@@ -164,6 +164,8 @@ func (a *API) buildDefinitionAPI() handlerBuilder {
 
 		adlV1 := r.Group("template/adl1.4")
 		adlV1.GET("/:template_id", a.Template.GetByID)
+		adlV1.GET("", a.Template.ListStored)
+		adlV1.POST("", a.Template.Store)
```
Comment (SealTV) [inline src/pkg/api/api.go:168]: maby here you to paste `"/"`

Comment (SealTV) [inline src/pkg/api/api.go:168]: maby here you to paste `"/"`

File: src/pkg/api/template.go [hunk 2]
```diff
@@ -2,19 +2,23 @@ package api
...
 	"hms/gateway/pkg/docs/model"
 	"hms/gateway/pkg/docs/service/template"
+	"hms/gateway/pkg/errors"
 )
 
 type TemplateService interface {
 	Parser(version model.ADLVer) (template.ADLParser, error)
 	GetByID(ctx context.Context, userID string, templateID string) (*model.Template, error)
+	Store(ctx context.Context, userID, systemID, reqID string, m *model.Template) error
+	List(ctx context.Context, userID, systemID string) ([]*model.TemplateResponse, error)
```
Comment (SealTV) [inline src/pkg/api/template.go]: It's a good practice to name functions as `verb` which describe action that it does (ex `GetParser` and `GetList`)

**Review by SealTV — APPROVED** [review]
left some comments

**Review by 1db1 — APPROVED** [review]

**Pull request merged**